### PR TITLE
[MRG] Add mistralai support

### PR DESCRIPTION
**Pull request by suryavanshi**
- Added support for MistralAI SDK
 - The Mistral model (test with - mistral-large-latest) doesn't always return json, so also added a function to use regex to get proper json.
 - Also found a bug here - https://github.com/MLSysOps/MLE-agent/blob/8109d8b905bb4aac4de6258516a966abaf2147f5/mle/agents/advisor.py#L188 - currently when providing a dataset its getting set to 'None'

### CODE REVIEW

**Comment by HuaizhengZhang** [comment]
wow. thanks for your contribution. expect to merge it.

ping us and add `[MRG]` if your pr is ready to be reviewed.

File: mle/agents/advisor.py [hunk 1]
```diff
@@ -102,6 +103,14 @@ def __init__(self, model, console=None):
         self.sys_prompt += self.json_mode_prompt
         self.chat_history.append({"role": 'system', "content": self.sys_prompt})
 
+    def clean_json_string(self, input_string):
+        cleaned = input_string.strip()
+        cleaned = re.sub(r'^```\s*json?\s*', '', cleaned)
+        cleaned = re.sub(r'\s*```\s*$', '', cleaned)
+        print("Cleaned:",cleaned)
+        parsed_json = json.loads(cleaned)
+        return parsed_json
```
Comment (leeeizhang) [inline mle/agents/advisor.py:112]: Many thanks! It appears that Mistral now officially supports `response_format='json'` (https://docs.mistral.ai/capabilities/json_mode/), so is it still necessary to manually clean the JSON string?

**Comment by huangyz0918** [comment]
Would you like to follow this up? @leeeizhang

**Comment by leeeizhang** [comment]
> Would you like to follow this up? @leeeizhang 

yes, i will improve some details and test it soon.

File: mle/agents/advisor.py (hunk 1 above)
Comment (suryavanshi) [reply mle/agents/advisor.py:112]: So the code here is already using that - https://github.com/MLSysOps/MLE-agent/blob/13968d7c79807ebcd0a95e3fdb7d23b06805dcbb/mle/agents/advisor.py#L117 - but the model is still not returning only json.

**Comment by suryavanshi** [comment]
> wow. thanks for your contribution. expect to merge it.
> 
> ping us and add `[MRG]` if your pr is ready to be reviewed.

Thanks, I think mistral json mode doesn't always work, so looking into it.

**Review by leeeizhang — APPROVED** [review]
LGTM

**Pull request merged**